components/server/ee/src/prebuilds/prebuild-manager.ts

```diff
@@ -58,6 +58,22 @@ export class PrebuildManager {
     @inject(Config) protected readonly config: Config;
     @inject(ProjectsService) protected readonly projectService: ProjectsService;
 
+    async abortPrebuildsForBranch(ctx: TraceContext, project: Project, user: User, branch: string): Promise<void> {
+        const span = TraceContext.startSpan("abortPrebuildsForBranch", ctx);
+        const prebuilds = await this.workspaceDB
+            .trace({ span })
+            .findActivePrebuiltWorkspacesByBranch(project.id, branch);
+        const results: Promise<any>[] = [];
+        for (const prebuild of prebuilds) {
+            for (const instance of prebuild.instances) {
+                results.push(this.workspaceStarter.stopWorkspaceInstance({ span }, instance.id, instance.region));
```

Comment:
I wonder what happens for instances which are not in a "stoppable" state, e.g. in image build? Will this fail for some not-yet-stoppable instances, or will this wait until it reaches a stoppable state? (I'm guessing the former.)